include/random_walks/uniform_accelerated_billiard_walk.hpp

```diff
@@ -127,30 +239,52 @@ struct AcceleratedBilliardWalk
                 }
 
                 _lambda_prev = dl * pbpair.first;
-                _p += (_lambda_prev * _v);
+                if constexpr (std::is_same<MT, Eigen::SparseMatrix<NT, Eigen::RowMajor>>::value) {
+                    _update_parameters.moved_dist = _lambda_prev;
+                    typename Point::Coeff b = P.get_vec();
+                    NT* b_data = b.data();
+                    NT* Ar_data = _lambdas.data();
+                    NT* Av_data = _Av.data();
+                    for(int i = 0; i < P.num_of_hyperplanes(); ++i) {
+                        _distances_set.vec[i].first = ( *(b_data + i) - (*(Ar_data + i)) ) / (*(Av_data + i));
+                    }
+                    _distances_set.rebuild(_update_parameters.moved_dist);
+                } else {
+                    _p += (_lambda_prev * _v);
+                }
                 T -= _lambda_prev;
                 P.compute_reflection(_v, _p, _update_parameters);
                 it++;
 
                 while (it < _rho)
-                {
-                    std::pair<NT, int> pbpair
-                            = P.line_positive_intersect(_p, _v, _lambdas, _Av, _lambda_prev,
-                                                        _AA, _update_parameters);
+                {   
+                    std::pair<NT, int> pbpair;
+                    if constexpr (std::is_same<MT, Eigen::SparseMatrix<NT, Eigen::RowMajor>>::value) {
+                        pbpair = P.line_positive_intersect(_p, _lambdas, _Av, _lambda_prev,
+                                                           _distances_set, _AA, _update_parameters);
+                    } else {
+                        pbpair = P.line_positive_intersect(_p, _v, _lambdas, _Av, _lambda_prev,
+                                                           _AA, _update_parameters);
+                    }
                     if (T <= pbpair.first) {
                         _p += (T * _v);
                         _lambda_prev = T;
                         break;
                     }
                     _lambda_prev = dl * pbpair.first;
-                    _p += (_lambda_prev * _v);
+                    if constexpr (std::is_same<MT, Eigen::SparseMatrix<NT, Eigen::RowMajor>>::value) {
+                        _update_parameters.moved_dist += _lambda_prev;
+                    } else {
+                        _p += (_lambda_prev * _v);
+                    }
                     T -= _lambda_prev;
                     P.compute_reflection(_v, _p, _update_parameters);
                     it++;
                 }
+                _p += _update_parameters.moved_dist * _v;
+                _update_parameters.moved_dist = 0.0;
                 if (it == _rho) {
                     _p = p0;
-                    was_reset = true;
```

Comment:
Why we don't need this flag anymore?

Reply:
basically for the sparse computations I needed to always use line_first_positive_intersect, so if I kept the was_reset flag I'd have to do sth like if(was_reset || is_sparse) use line_first_positive_intersect, else use line_positive_intersect. And while this might seem like it makes dense computations less efficient, in my opinion it barely makes any difference since the only difference between two functions is this
![image](https://github.com/user-attachments/assets/0b9e6bd0-5fd1-4082-8d46-ada5221a0845)
![image](https://github.com/user-attachments/assets/25997f7b-3a80-4e63-9112-87668085d6b8)
which is O(nm) anyways. If you think it matters I could bring it back and maybe think of some clean way to adapt it for the sparse case, but for now this was the easiest solution I found.

Reply:
Thanks!